src/scenes/gameuiscreen.ts

```diff
@@ -100,6 +113,69 @@ export class GameUiScreen extends Phaser.Scene {
         );
     }
 
+    public displayProfiles(): void {
+        const allPlayers: Nullable<Player[]> = PlayerManager.getAll();
+        const me: Nullable<Player> = PlayerManager.getMe();
+        const session: Nullable<Session> = SessionManager.get();
+        assert(me && allPlayers && session && this.profilesContainer);
+        this.profilesContainer.removeAll(true);
+        const playerCount: int = allPlayers.length;
+
+        const profileSpacing: int =
+            (ScreenWidth - playerCount * GameUiScreen.profilePixels) /
+            (playerCount + 1);
+        const y: float = ScreenHeight / 10;
+
+        for (let i: int = 0; i < playerCount; i++) {
+            let name: string = allPlayers[i].name;
+            const x: float =
+                profileSpacing +
+                GameUiScreen.profilePixels / 2 +
+                i * (GameUiScreen.profilePixels + profileSpacing);
+
+            const profile = this.add.image(
+                x,
+                y,
+                `profile${allPlayers[i].profile}`,
+            );
+            if (SessionManager.isPlayersTurn(allPlayers[i], playerCount)) {
+                this.profilesContainer.add(this.add.image(x, y, "ring"));
+            }
+            if (allPlayers[i].orderIndex === me.orderIndex) {
+                name += " (Me)";
+                let role: string = "Miner";
+                if (me.role === Role.Saboteur) {
+                    role = "Saboteur";
+                }
+                const roleText = this.add.text(
+                    x - (role.length * 6.5) / 2,
+                    y - GameUiScreen.profilePixels,
+                    role,
+                    {
+                        fontFamily: "Verdana",
+                        fontSize: "11px",
+                        color: "#ffd700",
+                        fontStyle: "bold",
+                    },
+                );
+                this.profilesContainer.add(roleText);
+            }
+            const nameText = this.add.text(
+                x - (name.length * 6) / 2,
+                y + GameUiScreen.profilePixels / 1.5,
+                name,
+                {
+                    fontFamily: "Verdana",
+                    fontSize: "11px",
+                    color: "#549b3d",
+                    fontStyle: "bold",
+                },
+            );
+            this.profilesContainer.add(nameText);
+            this.profilesContainer.add(profile);
+        }
+    }
```

Comment:
I like it but this function is way to big and you should refactor it into multiple smaller ones.